libfwupdplugin/fu-context.c

```diff
@@ -1619,6 +1619,20 @@ fu_context_get_esp_volumes(FuContext *self, GError **error)
 	return g_ptr_array_ref(priv->esp_volumes);
 }
 
+static gboolean
+fu_context_is_esp(FuVolume *esp)
+{
+	g_autofree gchar *mount_point = fu_volume_get_mount_point(esp);
+	g_autofree gchar *fn = NULL;
+
+	if (mount_point == NULL)
+		return FALSE;
+
+	fn = g_build_filename(mount_point, "EFI", NULL);
```

Comment:
the uppercase EFI does worry me here; does this always work on all filesystems?

Reply:
Don't we only support ESP on vfat _anyway_?  I did check with a case insensitive version on my SUT and it works.
```
	fn = g_build_filename(mount_point, "efi", NULL);
```

```
mount | grep /boot
/dev/nvme0n1p1 on /boot type vfat (rw,relatime,fmask=0077,dmask=0077,codepage=437,iocharset=ascii,shortname=mixed,utf8,errors=remount-ro)
sudo tree /boot/
/boot/
├── 831b88275656487cafb1d0a43bcbfad0
│   └── 6.9.2-2-cachyos
│       ├── initrd
│       └── linux
├── amd-ucode.img
├── EFI
│   ├── BOOT
│   │   └── BOOTX64.EFI
│   ├── Linux
│   ├── systemd
│   │   ├── fw
│   │   └── systemd-bootx64.efi
│   └── UpdateCapsule

11 directories, 20 files